utils/ESR_Peers_List.cpp

```diff
@@ -0,0 +1,317 @@
+    }
+
+
+    std::string
+    hex_to_utf8(const std::string& hex)
+    {
+        std::stringstream utft8stream;
+        boost::algorithm::unhex(hex, std::ostream_iterator<char>{utft8stream, ""});
+        return utft8stream.str();
+    }
+
+
+    std::string
+    string_to_hex(const std::string& value)
+    {
+        std::stringstream hexstream;
+        boost::algorithm::hex(value, std::ostream_iterator<char>{hexstream, ""});
+        return hexstream.str();
+    }
+
+
+    std::vector<std::string>
+    get_peers_result_to_vector(const std::string_view &result)
+    {
+        std::vector<std::string> results;
+        std::vector<std::string> lines;
+        size_t node_count{0};
+        enum states {HEADER, HEADER_SWARM_SIZE, HEADER_INFO, PEER_ID_SIZE, PEER_ID } state{HEADER};
+
+        char line[65]{0};
+        size_t index{0};
+        std::string peer_id;
+
+        std::istringstream stm(result.begin());
+
+        size_t peer_id_length{0};
+        while (stm.read(line, 64)) {
+            switch (state) {
+                case HEADER:
+                    state = HEADER_SWARM_SIZE;
+                    break;
+                case HEADER_SWARM_SIZE:
+                    state = HEADER_INFO;
+                    node_count = std::strtoul(line, nullptr, 16);
+                    if (node_count==0)
+                    {
+                        LOG(error) << "Requested swarm may not exist or has no nodes";
+                        return results;
+                    }
+                    break;
+                case HEADER_INFO:
+                    state = index == node_count+1 ? PEER_ID_SIZE : state ;
+                    break;
+                case PEER_ID_SIZE:
+                    // the first line of data is the size of the string
+                    peer_id_length = std::strtoul(line, nullptr, 16);
+                    state = PEER_ID;
+                    break;
+                case PEER_ID:
+                    peer_id.append(hex_to_utf8(line));
+                    if (peer_id.size() >= peer_id_length)
+                    {
+                        trim_right_nulls(peer_id);
+                        results.emplace_back(peer_id);
+                        peer_id = "";
+                        state = PEER_ID_SIZE;
+                    }
+                default:
+                    break;
+            }
+            index++;
+        }
+        assert(results.size()==node_count);
+        return results;
+    }
+
+
+    bzn::peer_address_t
+    get_peer_info_result_to_peer_address(const std::string& peer_id, const std::string_view &result)
+    {
+        uint16_t    port{0};
+        std::string host;
+        uint16_t    http_port{0};
+        std::string name;
+        enum {NODE_COUNT, NA_0, HTTP_PORT, NA_1, NODE_PORT, NA_2, NODE_HOST, NA_3, NODE_NAME} state {NODE_COUNT};
+
+        char line[65]{0};
+        std::istringstream stream{result.begin()};
+        while(stream.read(line, 64))
```

Comment:
same in this func

Reply:
Right, corrected the 65 and made the 64 a constant.